src/Microsoft.ML.PCA/PcaTrainer.cs

```diff
@@ -103,23 +103,23 @@ public class Arguments : UnsupervisedLearnerInputBaseWithWeight
 
         }
 
-        internal RandomizedPcaTrainer(IHostEnvironment env, Arguments args)
-            :this(env, args, args.FeatureColumn, args.WeightColumn)
+        internal RandomizedPcaTrainer(IHostEnvironment env, Options options)
```

Comment:
>RandomizedPcaTrainer [](start = 17, length = 20)

It's strange... I noticed that renaming Arguments to Options did not modify anything in the mlContext catalog.

Reply:
I looked it up, and I don't think there is an entry for this trainer in mlContext. Can you add it?

---
In reply to: [253319239](https://github.com/dotnet/machinelearning/pull/2390#discussion_r253319239) [](ancestors = 253319239)

Reply:
yeah.  i noticed couple more components which do not have mlcontext extension. 

will add

---
In reply to: [253319255](https://github.com/dotnet/machinelearning/pull/2390#discussion_r253319255) [](ancestors = 253319255,253319239)

Reply:
i added mlcontext  extension for this.  Also added a test  for it that exercises the Fit() and Transform()  APIs.

 Evaluate() API currently missing from Anomaly Detection. i will create a separate issue for that.

---
In reply to: [253584603](https://github.com/dotnet/machinelearning/pull/2390#discussion_r253584603) [](ancestors = 253584603,253319255,253319239)